src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1183,3 +1214,30 @@ export class CustomTreeView extends TreeView {
 		}
 	}
 }
+
+export class CustomTreeViewDragAndDrop implements ITreeDragAndDrop<ITreeItem> {
+	constructor(private dataSource: TreeDataSource) { }
+
+	onDragOver(data: IDragAndDropData, targetElement: ITreeItem, targetIndex: number, originalEvent: DragEvent): boolean | ITreeDragOverReaction {
+		return { accept: true, bubble: TreeDragOverBubble.Down, autoExpand: true };
+	}
+
+	getDragURI(element: ITreeItem): string | null {
+		return element.tooltip ? element.tooltip.toString() : null;
+	}
+
+	getDragLabel?(elements: ITreeItem[]): string | undefined {
+		if (elements.length > 1) {
+			return String(elements.length);
+		}
+		const element = elements[0];
+		return element.label ? element.label.label : undefined;
+	}
+
+	async drop(data: IDragAndDropData, targetNode: ITreeItem | undefined, targetIndex: number | undefined, originalEvent: DragEvent): Promise<void> {
+		const elements = (data as ElementsDragAndDropData<ITreeItem>).elements;
```

Comment:
I think you might need to do a check for `ElementsDradAndDropData` here instead of casting it. I suspect that `drop` can get called when something outside the treeview is dragged in. Like as is done here:

https://github.com/barbaravaldez/vscode/blob/731402dce225ce25d42a7b173361650a54bf52ad/src/vs/workbench/contrib/files/browser/views/openEditorsView.ts#L690-L691